src/metrics.jl

```diff
@@ -12,13 +12,15 @@ abstract type UnionSemiMetric <: SemiMetric end
 
 abstract type UnionMetric <: Metric end
 
+abstract type UnionMinkowskiMetric <: MinkowskiMetric end
+
```

Comment:
What do we need this for? Can't we replace any usage of `UnionMinkowskiMetric` by `MinkowskiMetric`? Users could simply subtype `MinkowskiMetric`, which is already abstract.

Reply:
I created this type to follow the pattern of the other "Union" types:
```
abstract type UnionPreMetric <: PreMetric end

abstract type UnionSemiMetric <: SemiMetric end

abstract type UnionMetric <: Metric end
```

From what I understand, these "Union" types are treated differently in the implementation.

However, since all subtypes of `MinkowskiMetric` were subtypes of `UnionMetric`, it is possible to replace `UnionMinkowskiMetric` with `MinkowskiMetric` in the code.

Reply:
It's been a very long while when I was looking at/dealing with these things, but these types are needed for multiple dispatch AFAIR. They used to be just `Union`s, but then people couldn't reuse their infrastructure. Basically, the corresponding *metrics have the same computational pattern, just different operations in the details.

Couldn't you just subtype `MinkowskiMetric <: UnionMetric`? Thinking about it more, I think you actually should do that, because otherwise these types fall out of the `UnionMetric` type branch, which potentially other users rely on. You would then simply introduce another type layer, to distinguish some metrics (`MinkowskiMetric`) from other `UnionMetric` metrics.

Reply:
So, is the PR code correct? Should we keep `UnionMinkowskiMetric`?

Reply:
Done!